cardano-api/src/Cardano/Api/ScriptData.hs

```diff
@@ -131,21 +136,30 @@ instance ToCBOR ScriptData where
 instance FromCBOR ScriptData where
   fromCBOR = fromPlutusData <$> decode @Plutus.Data
 
-hashScriptData :: ScriptData -> Hash ScriptData
+type HashableScriptData = WithCBOR ScriptData
+
+pattern AsHashableScriptData :: AsType (WithCBOR ScriptData)
+pattern AsHashableScriptData = AsWithCBOR AsScriptData
+
+hashScriptData :: HashableScriptData -> Hash ScriptData
```

Comment:
I'm not sure if this is needed any more, I think we can just go with `hashAnnotated` from Cardano.Ledger.SafeHash